src/useChat.tsx

```diff
@@ -32,24 +37,25 @@ export const useChat = (initValues?: ChatMessage[], initModel?: string) => {
     errorStore: new TextStreamStore(),
   });
 
-  useEffect(() => {
-    const id = setInterval(() => {
-      setConversationHistories((prev) => {
-        const storedConversations = JSON.parse(
-          localStorage.getItem('conversations') ?? '{}',
-        ) as ConversationHistories;
-
-        if (JSON.stringify(prev) !== JSON.stringify(storedConversations)) {
-          return storedConversations;
-        }
-        return prev;
+  const fetchConversations = useCallback(() => {
+    getAllConversations()
+      .then((conversations) => {
+        setConversationHistories(
+          conversations.reduce((acc, cur) => {
+            acc[cur.id] = cur;
+            return acc;
+          }, {} as ConversationHistories),
+        );
```

Comment:
Once conversations is an object, then you won't need to recreate it here

https://github.com/Kava-Labs/hard-ai/blob/646afb9c266270fcd1f6cd6e4573f1710453a396/src/useChat.tsx#L18